### PR TITLE
Clear out exceptionValue in branches where it's handled

### PR DESCRIPTION
**Pull request by elliottt**
Mark the exception as handled by writing `nil` to its `exceptionValue`. Additionally, this means we don't need to invent a variable that detects if the exception has been handled, as we can just use `exceptionValue` for that purpose.

### Motivation

Simplifying exception handling CFG.

### Test plan

See included automated tests.

### CODE REVIEW

File: cfg/Instructions.cc [hunk 1]
```diff
@@ -169,7 +169,7 @@ string LoadYieldParams::showRaw(const core::GlobalState &gs, int tabs) const {
 }
 
 string GetCurrentException::toString(const core::GlobalState &gs) const {
-    return "<unanalyzable>";
+    return "<get-current-exception>";
```
Comment (jez) [inline cfg/Instructions.cc:172]: 🎉

**Comment by elliottt** [comment]
This raised some errors on the stripe monorepo: http://go/builds/bui_HHpUivngebTLLC

Looking through them I think they're valid, so I'll fix them on Monday before merging this.

**Comment by elliottt** [comment]
@DarkDimius could you take another look? The problems that showed up on the stripe monorepo were from this change, and I've rewritten it to avoid them.

**Comment by elliottt** [comment]
Internal stripe build: http://go/builds/bui_HIpPOxi2pgv6eE

**Pull request merged**